src/sentry/rules/processing/delayed_processing.py

```diff
@@ -116,7 +116,7 @@ def get_condition_groups(
                 # set of group_ids that apply to the unique condition
                 else:
                     condition_groups[unique_condition] = DataAndGroups(
-                        condition_data, rules_to_groups[rule.id]
+                        condition_data, set(rules_to_groups[rule.id])
```

Comment:
For anyone reading, just pointing out that this is the fix - we copy the set here to avoid polluting the data